services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_watchdog_extentions.py

```diff
@@ -0,0 +1,54 @@
+import os
+from functools import reduce
+
+from watchdog.observers.api import DEFAULT_OBSERVER_TIMEOUT, BaseObserver
+from watchdog.observers.inotify import InotifyBuffer, InotifyEmitter
+from watchdog.observers.inotify_c import Inotify, InotifyConstants
+from watchdog.utils import BaseThread
+from watchdog.utils.delayed_queue import DelayedQueue
+
+_EVENTS_TO_WATCH = reduce(
+    lambda x, y: x | y,
+    [
+        InotifyConstants.IN_MODIFY,
+        InotifyConstants.IN_MOVED_FROM,
+        InotifyConstants.IN_MOVED_TO,
+        InotifyConstants.IN_CREATE,
+        InotifyConstants.IN_DELETE,
+        InotifyConstants.IN_DELETE_SELF,
+        InotifyConstants.IN_DONT_FOLLOW,
+        InotifyConstants.IN_CLOSE_WRITE,
+    ],
+)
+
+
+class _ExtendedInotifyBuffer(InotifyBuffer):
+    def __init__(self, path, recursive=False):  # pylint:disable=super-init-not-called
+        BaseThread.__init__(self)  # pylint:disable=non-parent-init-called
```

Comment:
As pylint warns, this is indeed a strange setup. I [see](https://github.com/gorakhargosh/watchdog/blob/78adc2f8b61858be28e8056b7a88f4a4ae7acab7/src/watchdog/observers/inotify_buffer.py) that you actually want to only override ``InotifyBuffer.__init__`` in order to change ``self._inotify`` before it ``self.start()``.
i wonder if there is a better way using ``__new__`` ... i need to think about it and will come back to you

Reply:
Yes, the internals of this library do not give me much flexibility. This was the cleanest way to do it. I'd appreciate further input here, on how to make make it more clean.